src/Components/Order/OrderInfo.tsx

```diff
@@ -1,213 +1,165 @@
-  width: 100%;
-  height: 44px;
-  padding: 0 12px;
-  font-size: 14px;
-  border: 1px solid ${({ error, theme }) => (error ? 'red' : theme.color.semantic.border.default)};
-  border-radius: 8px;
-  outline: none;
-
-  &::placeholder {
-    color: ${({ theme }) => theme.color.semantic.text.placeholder};
-  }
-`;
-
-const ErrorMsg = styled.p`
-  color: ${({ theme }) => theme.color.semantic.status.critical};
-  font-size: 12px;
-  margin: 0;
-`;
-
-const HelperText = styled.span`
-  font-size: 12px;
-  color: ${({ theme }) => theme.color.semantic.text.placeholder};
-`;
-
-const ProductInfo = styled.div`
-  padding: 20px 16px;
-  border-radius: 8px;
-  display: flex;
-  gap: 12px;
-  align-items: center;
-`;
-
-const ProductImage = styled.img`
-  width: 64px;
-  height: 64px;
-  object-fit: cover;
-  border-radius: 4px;
-`;
-
-const ProductDetails = styled.div`
-  display: flex;
-  flex-direction: column;
-  gap: 4px;
-
-  & > strong {
-    font-size: 14px;
-  }
-
-  & > span {
-    font-size: 12px;
-    color: ${({ theme }) => theme.color.semantic.text.placeholder};
-  }
-
-  & > b {
-    font-weight: bold;
-  }
-`;
-
-const OrderButton = styled.button`
-  width: 100%;
-  height: 56px;
-  background-color: ${({ theme }) => theme.color.semantic.brand.kakaoYellow};
-  border: none;
-  font-size: 16px;
-  font-weight: bold;
-  border-radius: 4px;
-  cursor: pointer;
-`;
-
-const GiftForm = () => {
+import type { Receiver } from '@/types/receiver';
+import {
+  Wrapper,
+  Section,
+  Label,
+  InputBox,
+  StyledInput,
+  StyledTextarea,
+  ErrorMsg,
+  HelperText,
+  ProductInfo,
+  ProductImage,
+  ProductDetails,
+  OrderButton,
+} from '@/components/Order/Order.style';
+
+interface GiftFormValues {
+  sender: string;
+  message: string;
+}
+
+interface GiftSenderProps {
+  templateMessage: string;
+}
+
+const GiftForm = ({ templateMessage }: GiftSenderProps) => {
   const navigate = useNavigate();
   const location = useLocation();
   const giftId = location.state?.id;
-
   const selectedGift = GiftList.find((gift) => gift.id === giftId);
 
-  const { values, errors, handleChange, validate } = useOrderForm({
-    sender: '',
-    receiver: '',
-    phone: '',
-    quantity: 1,
-    message: '',
+  const [receiverList, setReceiverList] = useState<Receiver[]>([]);
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    formState: { errors },
+  } = useForm<GiftFormValues>({
+    defaultValues: {
+      sender: '',
+      message: templateMessage ?? '',
+    },
   });
 
-  if (!selectedGift) {
-    return <div>선택한 상품이 없습니다.</div>;
-  }
+  useEffect(() => {
+    setValue('message', templateMessage);
+  }, [templateMessage, setValue]);
+
+  const validateReceivers = () => {
+    if (receiverList.length === 0) {
+      alert('최소 1명의 받는 사람을 등록해주세요.');
+      return false;
+    }
+
+    const phoneSet = new Set<string>();
+    for (let i = 0; i < receiverList.length; i++) {
+      const r = receiverList[i];
```

Comment:
```suggestion
    for (const [i, r] of receiverList.entries()) {
```
요렇게 하시면 가독성이 좀더 좋아질 것 같네요.